src/dafny/Relations.dfy

```diff
@@ -26,6 +26,10 @@ module {:options "-functionSyntax:4"} Dafny.Relations {
     forall x, y :: R(x, y) && R(y, x) ==> x == y
   }
 
+  ghost predicate Symmetric<T(!new)>(R: (T, T) -> bool) {
+    forall x, y :: R(x, y) <==> R(y, x)
```

Comment:
Just a question for my own education, what's the difference between using `<==>` and `==` here?

Reply:
According to Section 5.2.1.1. in https://dafny.org/dafny/DafnyRef/DafnyRef, _the expressions `A <==> B` and `A == B` give the same value, but note that `<==>` is associative whereas `==` is chaining and they have different precedence_.